### ISSUE TITLE
AuthorizationException thrown instead of ThrottleException

### ISSUE THREAD

**Issue by aarondfrancis** [facebook/facebook-php-ads-sdk#192]
During my testing, I hit the limit for a certain request and an `AuthorizationException` was thrown with the message `(#17) User request limit reached.`. 

I had seen that there was a `ThrottleException` and thought that would have been a better choice, so I went to see why that had not been thrown. I ended up looking at [RequestException, L131](https://github.com/facebook/facebook-php-ads-sdk/blob/2.5.1/src/FacebookAds/Http/Exception/RequestException.php#L131). I dumped my `$error_data` and got this:
```
array:7 [
  "code" => 17
  "error_subcode" => null
  "message" => "(#17) User request limit reached"
  "error_user_title" => null
  "error_user_msg" => null
  "error_blame_field_specs" => null
  "type" => "OAuthException"
]
```

So it looks like this error matches the first (complicated) condition of the `if` statement because the `type` is `OAuthException`. However, if you go down to the 2nd `elseif` you'll see that it's checking for a `code` of 17, which would throw the correct `ThrottleException`.

I wonder if the first `if` condition is too strict, or if perhaps the checking should be reversed to check for the more specific conditions (`ThrottleException`, etc) first, before just throwing the general `AuthorizationException`.

**Comment by duliomatos** [comment]
Thx for bringing this to our attention @aarondfrancis. We are relying on error type OAuthException being more correct than the `code` field, so I'll probably just reorder the conditions.

**Comment by duliomatos** [comment]
@aarondfrancis, I'm queueing this to be fixed on version 2.6. This is a breaking change for the sdk, that's why we wait for a major release to change behavior that developers may be relying upon.

**Comment by clin407** [comment]
I think this issue still exists. I just hit the limit and got this as the response:

{"error":{"message":"(#17) User request limit reached","type":"OAuthException","is_transient":true,"code":17,"fbtrace_id":"BxJ2rGNio0B"}}.

In my case, based off reading the rate limiting best practices, sounds like I hit the rate limit which should throw a ThrottleException.
